### PR TITLE
Preparation for the "Byron;Shelley" composite protocol

### PR DESCRIPTION
**Pull request by dcoutts**
Update to latest network and consensus.

Add generic tracing instances for the hard fork protocol combinator that we will need.

### CODE REVIEW

File: cardano-config/src/Cardano/TracingOrphanInstances/Common.hs [hunk 1]
```diff
@@ -57,3 +60,7 @@ import           Cardano.BM.Data.Tracer
 --
 instance ToObject Void where
   toObject _verb x = case x of {}
+
+
+--TODO: move this instance to the definition
+deriving instance Ord PrivacyAnnotation
```
Comment (dcoutts) [inline cardano-config/src/Cardano/TracingOrphanInstances/Common.hs:66]: https://github.com/input-output-hk/iohk-monitoring-framework/pull/567

File: cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+--
+-- instances for GenTx HardForkBlock
+--
+
+instance Condense (GenTx (HardForkBlock xs)) where
+  condense = error "TODO: replace with versions from upstream"
+
+instance Condense (TxId (GenTx (HardForkBlock xs))) where
+  condense = error "TODO: replace with versions from upstream"
```
Comment (dcoutts) [inline cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:79]: These are getting added to `ouroboros-consensus`

File: cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+-- instances for GenTx HardForkBlock
+--
+
+{-
+instance HasKESMetricsData (HardForkBlock xs) where
+    getKESMetricsData :: ProtocolInfo m (HardForkBlock xs)
+                      -> ForgeState (HardForkBlock xs)
+                      -> KESMetricsData
+    getKESMetricsData protoInfo forgeState =
+-}
```
Comment (dcoutts) [inline cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:109]: @mrBliss I could do with help on this one. For the `ForgeState` it's a product, but each one will expect a `ProtocolInfo` projected to it's individual block type.

File: cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:79]: Oh, in fact these are now upstream. Fixing...

File: cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs (hunk 3 above)
Comment (mrBliss) [reply cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:103]: Why does it need the whole `ProtocolInfo`? Writing this projection (let alone generically) will be impossible: for one, the `ProtocolInfo` contains the initial ledger state, but the `ProtocolInfo` for a `HardForkBlock` will only contain the initial ledger state for the first era, *not for all of them*.

A [quick search](https://github.com/input-output-hk/cardano-node/search?q=getKESMetricsData&unscoped_q=getKESMetricsData) revealed that the `ProtocolInfo` isn't actually used, so I would just remove it.

Comment (dcoutts) [reply cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:103]: Oh, nice. Ok, thanks.

Comment (dcoutts) [reply cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:103]: Turns out this is a bit harder. We can solve in a follow-up PR once more HFC stuff is merged in consensus.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**